pkg/sql/session.go

```diff
@@ -1381,17 +1383,58 @@ func (scc *schemaChangerCollection) execSchemaChanges(
 	return firstError
 }
 
+const panicLogOutputCutoffChars = 500
+
+// AnonymizeStatementsForReporting transforms an action, SQL statements, and a value
+// (usually a recovered panic) into an error that will be useful when passed to
+// our error reporting as it exposes a scrubbed version of the statements.
+func AnonymizeStatementsForReporting(action, sqlStmts string, r interface{}) error {
+	var anonymized []string
+	{
+		stmts, err := parser.Parse(sqlStmts)
+		if err == nil {
+			var buf bytes.Buffer
+			for _, stmt := range NewStatementList(stmts) {
+				tree.FormatNode(&buf, tree.FmtAnonymize, stmt.AST)
```

Comment:
Heh, can't you make a new FmtFlags that contains both FmtAnonymize and FmtHideConstants rather than parse and reformat each statement twice? Forgive me if I'm missing something that makes that necessary.